### ISSUE TITLE
@InheritConfiguration is broken when using multiple source parameters of the same type

### ISSUE THREAD

**Issue by bidorffOL** [mapstruct/mapstruct#2221]
- [X] Is this an issue (and hence not a question)? 

I have been trying to inherit a shared configuration for a multi-parameters mapper and I can't get it to work. My classes look like:

```java
@MapperConfig
public interface RestConfig {

  @Mapping(target = "tenantId", source = "tenantId")
  @Mapping(target = "siteId", source = "siteId")
  @Mapping(target = "ctiId", source = "source.cti", qualifiedBy = [...])
  SiteDto convert(RestSiteDto source, String tenantId, String siteId);

}

@Mapper(config = RestConfig.class)
public interface RestSiteMapper {

  @InheritConfiguration
  SiteDto convert(RestSiteDto source, String tenantId, String siteId);

}
```

However I keep getting the error : `Method has no source parameter named "tenantId". Method source parameters are: "source, tenantId, siteId".`

The weird thing is that if I only keep `@Mapping(target = "ctiId", source = "source.cti", qualifiedBy = [...])` the mapping works.

**Edit**: it seems to be a regression (it was working fine up until 1.3.1.Final), so I wonder if it is related to #2206 & #2214.

**Comment by filiphr** [comment]
Thanks for the report @bidorffOL. I don't think that it is related to #2206 and #2214. Those only have issues when no `source` is present.

**Comment by filiphr** [comment]
This was a tricky one @bidorffOL. The problem is that the source parameters in the template method are of the same type. There is a PR to fix this and it will be part of the 1.4.1 release.

**Comment by bidorffOL** [comment]
👍 That's great to hear, thank you !
